Source/MVC6/Boilerplate.AspNetCore.TagHelpers/Twitter/Cards/TwitterCardApp.cs

```diff
@@ -105,5 +101,33 @@ public override void ToString(StringBuilder stringBuilder)
             stringBuilder.AppendMetaNameContentIfNotNull("twitter:app:url:googleplay", this.GooglePlayCustomUrlScheme);
             stringBuilder.AppendMetaNameContentIfNotNull("twitter:app:country", this.Country);
         }
+
+        /// <summary>
+        /// Checks that this instance is valid and throws exceptions if not valid.
+        /// </summary>
+        protected override void Validate()
+        {
+            base.Validate();
+
+            if (string.IsNullOrEmpty(this.TwitterSiteUsername))
+            {
+                throw new ArgumentNullException(nameof(this.TwitterSiteUsername));
+            }
+
+            if (string.IsNullOrEmpty(this.IPhone))
```

Comment:
I believe some of these properties must be optional. We don't want to throw exceptions in that case.